GraphBase/src/graph/DirectedGraph.java

```diff
@@ -0,0 +1,49 @@
+package graph;
+
+public class DirectedGraph extends Graph {
+
+  public Vertex deleteVertex(int vertexIndex) {
+    Vertex deletedVertex = vertices.remove(vertexIndex);
+    if (deletedVertex != null) {
+      for (Integer key : vertices.keySet()) {
+        vertices.get(key).deleteAdjEdge(vertexIndex);
+      }
+    }
+    return deletedVertex;
+  }
+
+  public Vertex deleteVertex(Vertex vertex)  {
+    return deleteVertex(vertex.getId());
+  }
+
+  public int addEdge(int fromId, int toId) {
+    addVertex(fromId);
+    addVertex(toId);
+    if (!this.vertices.get(fromId).isAdjacent(toId)) {
+      Edge newEdge = new Edge(fromId, toId);
+      this.vertices.get(fromId).addFromAdjEdge(newEdge);
+      return 0;// 0 represents the edge is not existed , otherwise return 1;
```

Comment:
I think I commented for you to return boolean. Please make a change and you are Okay to go.